llpc/util/llpcFile.cpp

```diff
@@ -111,8 +111,11 @@ Result File::open(const char *filename, unsigned accessFlags) {
 
     if (result == Result::Success) {
 #if defined(_WIN32)
-      // MS compilers provide fopen_s, which is supposedly "safer" than traditional fopen.
-      fopen_s(&m_fileHandle, filename, &fileMode[0]);
+      // MS compilers provide _fsopen, which is supposedly "safer" than
```

Comment:
Does the "supposedly safer" comment still apply? I thought it was only for _s versions of standard library functions.

Reply:
Reading the docs... fopen_s is mainly safer because it prevents shareable access. I guess this one is less safe.
Perhaps it makes more sense to use the "_SH_DENYWR" option - this one doesn't prevent multiple read only accesses, but does stop anyone else writing to the file until we're done with it. That should fix the issue that's the problem in the testing, but still be safer in terms of concurrent write. (Then the comment can stay the same).

Reply:
+1. I think that the original comment might have referred to https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/fopen-wfopen?view=msvc-170:
> More-secure versions of these functions that perform additional parameter validation and return error codes are available; see fopen_s, _wfopen_s.

`_fsopen` does not return any error codes, so the comment does not seem to apply anymore

Reply:
I've re-worded it (I also tried using SH_DENYWR, but that still failed).